app/models/pame_evaluation.rb

```diff
@@ -83,62 +83,46 @@ def self.parse_filters(filters)
-        country_ids << countries.map{ |iso3| "#{ Country.find_by(iso_3: iso3).id }" }
-        where_params[:iso3] = country_ids.flatten.empty? ? nil : "countries.id IN (#{country_ids.join(',')})"
-      when 'methodology'
-        options = options.map{ |e| "'#{e}'" }
-        where_params[:methodology] = options.empty? ? nil : "methodology IN (#{options.join(',')})"
-      when 'year'
-        where_params[:year] = options.empty? ? nil : "year IN (#{options.join(',')})"
+    if filters.nil?
+      {}
+    else
+      filters.each do |filter|
+        options = filter["options"]
+        case filter['name']
+        when 'iso3'
+          countries = options
+          site_ids << countries.map{ |iso3| Country.find_by(iso_3: iso3).protected_areas.pluck(:id) }
+          where_params[:sites] = site_ids.flatten.empty? ? nil : "protected_area_id IN (#{site_ids.join(',')})"
+          country_ids << countries.map{ |iso3| "#{ Country.find_by(iso_3: iso3).id }" }
+          where_params[:iso3] = country_ids.flatten.empty? ? nil : "countries.id IN (#{country_ids.join(',')})"
+        when 'methodology'
+          options = options.map{ |e| "'#{e}'" }
+          where_params[:methodology] = options.empty? ? nil : "methodology IN (#{options.join(',')})"
+        when 'year'
+          where_params[:year] = options.empty? ? nil : "year IN (#{options.join(',')})"
+        end
       end
+      where_params
     end
-    where_params
   end
 
   def self.run_query(page, where_params)
     if where_params[:sites].present?
-      query = PameEvaluation.connection.unprepared_statement {
-        "((#{pame_evaluations_from_pa_query(where_params)}) UNION (#{pame_evaluations_from_countries_query(where_params)})) AS pame_evaluations"
-      }
-
       PameEvaluation
-      .from(query)
-      .paginate(page: page || 1, per_page: 100).order('id ASC')
+      .joins(:protected_area)
+      .where('protected_area_id IS NOT NULL')
+      .where(where_params[:sites])
+      .where(where_params[:methodology])
+      .where(where_params[:year])
     else
       PameEvaluation
       .where(where_params[:methodology])
       .where(where_params[:year])
-      .paginate(page: page || 1, per_page: 100).order('id ASC')
-    end
-  end
-
-  def self.pame_evaluations_from_pa_query(where_params)
-    PameEvaluation
-    .joins(:protected_area)
-    .where(where_params[:sites])
-    .where(where_params[:methodology])
-    .where(where_params[:year])
-    .to_sql
-  end
-
-  def self.pame_evaluations_from_countries_query(where_params)
-    PameEvaluation
-    .joins(:countries)
-    .where(where_params[:iso3])
-    .where(where_params[:methodology])
-    .where(where_params[:year])
-    .to_sql
+    end.paginate(page: page || 1, per_page: 100).order('id ASC')
   end
 
   def self.serialise(evaluations)
-    evaluations.select{|pe| pe.visible}.to_a.map! { |evaluation|
+    evaluations.select{|pe| pe.protected_area}.to_a.map! { |evaluation|
```

Comment:
Could you please use the `do` block instead of the curly brackets when the actual block is not a one liner?